icgcget/clients/gnos/gnos_client.py

```diff
@@ -17,33 +19,46 @@
 #
 
 import re
-import tempfile
+import os
 from icgcget.clients.download_client import DownloadClient
 from icgcget.clients.errors import SubprocessError
 
 
 class GnosDownloadClient(DownloadClient):
 
-    def __init__(self, json_path=None):
-        super(GnosDownloadClient, self) .__init__(json_path)
+    def __init__(self, json_path=None, docker=False):
+        super(GnosDownloadClient, self).__init__(json_path, docker)
         self.repo = 'cghub'
 
-    def download(self, uuids, access, tool_path, output, processes, udt=None, file_from=None, repo=None, password=None):
-        access_file = tempfile.NamedTemporaryFile()
-        access_file.file.write(access)
-        access_file.file.seek(0)
-        call_args = [tool_path, '-vv', '-c', access_file.name, '-d']
-        call_args.extend(uuids)
-        call_args.extend(['-p', output])
+    def download(self, uuids, access, tool_path, staging, processes, udt=None, file_from=None, repo=None,
+                 password=None):
+        access_file = self.get_access_file(access, staging)
+        call_args = []
+        if self.docker:
+            access_path = '/icgc/mnt/' + os.path.split(access_file.name)[1]
+            call_args = ['/bin/sh', '-c', tool_path + ' -vv ' + '-d ' + ' '.join(uuids) + ' -c ' + access_path +
```

Comment:
Please document why we are using `/bin/sh` (I can't recall, but I know you had issues)